src/widgets/index/index.ts

```diff
@@ -105,8 +106,12 @@ function resolveScopedResultsFromIndex(widget: Index): ScopedResult[] {
   return resolveScopedResultsFromWidgets(widgetSiblings);
 }
 
-const index = (props: IndexProps): Index => {
-  const { indexName = null } = props || {};
+const index = (props?: IndexProps): Index => {
```

Comment:
The switch to optional is not required we lose the type-safety. An alternative is to check the value even though the compiler ensure the value is always available. At least TypeScript users (mainly us) don't lose type-safety.

```ts
const index = (props: IndexProps): Index => {
  if (!props) {
    throw new Error(withUsage('Cannot be called without args.'));
  }

  const { indexName = null, indexId = indexName } = props;

  if (indexName === null) {
    throw new Error(withUsage('The `indexName` option is required.'));
  }

  if (indexId === null) {
    // Should not happen but at least the compiler is happy.
    throw new Error(withUsage("The `indexId` option can't be `null`."));
  }
   
  // ...
```

Reply:
will remove the ?

Reply:
db4f325c7